src/components/TimeCounter.tsx

```diff
@@ -1,79 +1,61 @@
-import { StackNavigationProp } from '@react-navigation/stack';
-import dayjs from 'dayjs';
-import React, { useState, useEffect } from 'react';
-import { StyleSheet, Text, View, KeyboardAvoidingView } from 'react-native';
-import { Overlay } from 'react-native-elements';
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
 
-import { MainCardStackNavigatorParams, Route } from 'app/consts';
-import { useInterval } from 'app/helpers/useInterval';
-import { typography } from 'app/styles';
+import { palette, fonts } from 'app/styles';
+import { secondsToFormat } from 'app/utils/date';
 
-import { Button } from './Button';
-
-const i18n = require('../../loc');
+import { Text } from './Text';
 
 interface Props {
-  timestamp: number;
-  isVisible: boolean;
-  navigation: StackNavigationProp<MainCardStackNavigatorParams, Route.CurrentPin>;
-  onCountFinish?: () => void;
+  value: number;
 }
 
-export const TimeCounter = ({ timestamp, onCountFinish, isVisible, navigation }: Props) => {
-  const currentTimestamp = dayjs().unix();
-  const secondsToCount = (timestamp - currentTimestamp).toFixed(0);
-  const [seconds, setSeconds] = useState(parseInt(secondsToCount));
-
-  const isTimeUp = (): boolean => seconds === 0;
-
-  useInterval(
-    () => {
-      setSeconds(seconds - 1);
-    },
-    seconds > 0 ? 1000 : null,
-  );
-
-  useEffect(() => {
-    if (isTimeUp()) {
-      onFinish();
-    }
-  });
-
-  const onFinish = () => {
-    if (onCountFinish) {
-      onCountFinish();
-    }
-  };
-
-  const goBack = () => navigation.goBack();
-
+export const TimeCounter = ({ value }: Props) => {
+  const middleElement = 2;
+  const time = secondsToFormat(value, 'mm:ss');
   return (
-    <Overlay isVisible={isVisible}>
-      <View style={styles.container}>
-        <View style={styles.timerContainer}>
-          <Text style={typography.headline4}>{i18n.onboarding.numberOfAttemptsExceeded}</Text>
-          <Text style={typography.headline4}>
-            {i18n.onboarding.tryAgain} {seconds} {i18n.onboarding.seconds}
-          </Text>
-        </View>
-        <KeyboardAvoidingView style={styles.footer} keyboardVerticalOffset={20}>
-          <Button onPress={goBack} title={i18n.onboarding.goBack} />
-        </KeyboardAvoidingView>
-      </View>
-    </Overlay>
+    <View style={styles.container}>
+      {time.split('').map((el, index) => {
+        const number = time[index];
+        if (index === middleElement) {
+          return (
+            <View key={index} style={styles.breakContainer}>
```

Comment:
those two are identical they have just diffrent styling so you can do <View key={index} style={[el === ':' ? styles.breakContainer: styles.valueContainer]}> and delete duplicate. 
Instead of hardcoding the position of ":" I would just check if element equals ":", this is more flexible as you don't have to care about the position of ":"